lib/interface/helpers.rb

```diff
@@ -2,17 +2,47 @@
 
 module Interface
   module Helpers
-    # Errors raised here identify the class_name that is not implementing
-    # the method required by the Interface.
+    # Errors raised here identify the class_name that is not implementing or is
+    # implemented with the wrong arity required by the Interface.
     def must_implement(*args)
-      args.each do |method_name|
-        next if respond_to? method_name
-        raise Interface::Error::NotImplementedError.new(
-                class_name: self.class.name,
-                method_name: method_name,
-                interface_name: self.class.ancestors.first
-              )
+      args.each do |method|
+        if method.is_a? Hash
```

Comment:
@rodrigo-ehresmann thanks! - one last thing, doesn't this work without checking if method is a hash? Doing something like this directly:

```ruby
def must_implement(*args)
  args.each do |method_name, arity|
    next if respond_to?(method_name) && expected_arity?(method_name, arity)

    raise...
  end
end

private

def expected_arity?(method_name, arity)
  return true unless arity

  method_arity(method_name) == arity
end
```

It will simplify things even more.